static/js/compose.js

```diff
@@ -565,7 +527,6 @@ function send_message(request) {
     if (request === undefined) {
         request = create_message_object();
     }
-    exports.snapshot_message(request);
```

Comment:
OK I tested this pretty carefully, including in the case where local echo is disabled, and merged
e608f09f737031384cf7713a1d5e97245263f2e4 for this.  I like to do a separate commit for this sort of thing, since this change is (hopefully) correct even independent from this PR, but has a pretty complex explanation for why that's the case :).